src/line_iterator.rs

```diff
@@ -64,9 +64,10 @@ impl<R: BufRead> Iterator for LineIterator<R> {
                         self.format = TldrFormat::V1;
                     } else {
                         // It's the new format! Drop next line.
-                        // (Hmm, is there a way to do this without an allocation?)
-                        let mut devnull = String::new();
-                        if let Err(e) = self.reader.read_line(&mut devnull) {
+                        if let Err(e) = Read::bytes(&mut self.reader)
+                            .find(|b| matches!(b, Ok(b'\n') | Err(_)))
+                            .transpose()
+                        {
```

Comment:
That looks quite elegant (and _much_ better than the original version 😄).

@niklasmohrin in your version, you mapped the result to a `()`, but as proposed by @qknogxxb I don't think we need that at all.

Furthermore, you used `<&mut R as std::io::Read>::bytes(...)` That doesn't seem necessary, as can be seen here, right? Or was it something else you had in mind?

Reply:
I was still writing with the function signature in mind, but inlining seems like a good call from @qknogxxb :)

The extra syntax was just to get around derefs, didn't notice that `Read::bytes` already picks up the correct impl. (`self.reader.by_ref().bytes()` still tries to use `BufReader::bytes`, not `(&mut BufReader)::bytes`)

Reply:
It seems that the compiler is not smart enough to infer how far it should to deref and see whether is it properly.
The compiler always deref to the end of the deref chain. A hint to compiler is necessary anyway.